dev/update_ml_package_versions.py

```diff
@@ -89,6 +90,48 @@ def update_max_version(src, key, new_max_version, category):
     return re.sub(pattern, rf'\g<1>"{new_max_version}"', src, flags=re.DOTALL)
 
 
+def extract_field(d, keys):
+    for key in keys:
+        if key in d:
+            d = d[key]
+        else:
+            return None
+    return d
+
+
+def generate_ml_package_versions_py(yaml_path):
+    with Path(yaml_path).open() as f:
+        config = {}
+        for name, cfg in yaml.load(f, Loader=yaml.SafeLoader).items():
+            # Extract required fields
+            pip_release = extract_field(cfg, ("package_info", "pip_release"))
+            min_version = extract_field(cfg, ("autologging", "minimum"))
+            max_version = extract_field(cfg, ("autologging", "maximum"))
+            if (pip_release, min_version, max_version).count(None) > 0:
+                continue
+
+            config[name] = {
+                "package_info": {
+                    "pip_release": pip_release,
+                },
+                "autologging": {
+                    "minimum": min_version,
+                    "maximum": max_version,
+                },
+            }
+
+        this_file = Path(__file__).resolve().relative_to(Path.cwd())
```

Comment:
Will this give the correct resolution on nt-based file systems?
I seem to recall I had issues with pathing resolution in windows  for functionality in a different feature last year and it had some odd behavior when trying to access `__file__` attributes depending on where it was called from.
IIRC (I may be wrong here) using `from inspect import getsourcefile` and mapping over a no-op returns the current executable source location on all OS's. 

Would something like this work?

```
import inspect
from os.path import abspath

this_file = Path(abspath(inspect.getsourcefile(lambda:0))).resolve().relative_to(Path.cwd())
```

Reply:
We won't run this script on Windows·

Reply:
sounds good :)